src/mondeath.cpp

```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
     }
     // make "broken_manhack", or "broken_eyebot", ...
     item_id.insert( 0, "broken_" );
-
     item broken_mon( item_id, calendar::turn );
     const int max_hp = std::max( z.get_hp_max(), 1 );
     const float overflow_damage = std::max( -z.get_hp(), 0 );
     const float corpse_damage = 2.5 * overflow_damage / max_hp;
     broken_mon.set_damage( static_cast<int>( std::floor( corpse_damage * itype::damage_scale ) ) );
 
     g->m.add_item_or_charges( z.pos(), broken_mon );
-
+	
+	
+    //adds ammo drop	
+    const bool drops_ammo = z.type->has_flag( MF_DROPS_AMMO ); 
+    if( drops_ammo ) {
+    for( const auto &ammo_entry : z.type->starting_ammo ) {
+		if( z.ammo[ammo_entry.first] > 0 ) {
+			g->m.spawn_item( z.pos(), ammo_entry.first.c_str(), z.ammo[ammo_entry.first.c_str()], 1, calendar::turn );
+		}
+	}
+}
+    //end adds ammo drop
```

Comment:
This is not needed. If you want to give a piece of code some name, put it into a function.